tst/EnergyPlus/unit/Fixtures/EnergyPlusFixture.hh

```diff
@@ -0,0 +1,194 @@
+		virtual void SetUp() {
+			show_message();
+			this->ioHelper = std::unique_ptr< IOHelper >( new IOHelper );
+		}
+
+		// This is run every unit test and makes sure to clear all state in global variables this fixture touches.
+		virtual void TearDown() {
+			CurveManager::clear_state();
+			DataEnvironment::clear_state();
+			DataGlobals::clear_state();
+			DataHeatBalance::clear_state();
+			DataIPShortCuts::clear_state();
+			DataOutputs::clear_state();
+			DataSurfaces::clear_state();
+			HeatBalanceManager::clear_state();
+			InputProcessor::clear_state();
+			OutputProcessor::clear_state();
+			ScheduleManager::clear_state();
+
+			{ 
+				IOFlags flags; 
+				flags.DISPOSE( "DELETE" );
+				gio::close( OutputProcessor::OutputFileMeterDetails, flags );
+				gio::close( DataGlobals::OutputFileStandard, flags ); 
+				gio::close( DataGlobals::OutputFileInits, flags );
+				gio::close( DataGlobals::OutputFileDebug, flags );
+				gio::close( DataGlobals::OutputFileZoneSizing, flags );
+				gio::close( DataGlobals::OutputFileSysSizing, flags );
+				gio::close( DataGlobals::OutputFileMeters, flags );
+				gio::close( DataGlobals::OutputFileBNDetails, flags );
+				gio::close( DataGlobals::OutputFileZonePulse, flags );
+
+			}
+		}
+
+		// This will output the "Begin Test" ShowMessage for every unit test that uses or inherits from this fixture.
+		// Now this does not need to be manually entered for every unit test as well as it will automatically be updated as the
+		// unit test names change.
+		void show_message() {
+			// Gets information about the currently running test.
+			// Do NOT delete the returned object - it's managed by the UnitTest class.
+			const ::testing::TestInfo* const test_info = ::testing::UnitTest::GetInstance()->current_test_info();
+			ShowMessage( "Begin Test: " + std::string( test_info->test_case_name() ) + ", " + std::string( test_info->name() ) );
+		}
+
+		static void use_cached_idd() {
+			IOHelper::use_cache();
+		}
+
+		// This will compare either a STL container or ObjexxFCL container
+		// Pass a forward iterator of a container you want to compare against an expected container. You can pass in an existing 
+		// container or use an initializer list like below.
+		// I had to only do one template typename due to wrapping this in Google Test macros causes the macro to fail. It thinks
+		// there are more macro arguments than expected. This is a way around that restriction.
+		// Example Usage: 
+		// 		EXPECT_TRUE( compare_containers< std::vector< bool > >( { true } , ObjectDef( index ).AlphaOrNumeric ) );
+		// 		EXPECT_TRUE( compare_containers< Array1D_bool >( { true } , ObjectDef( index ).AlphaOrNumeric ) );
+		template < typename T, class T2 >
+		bool compare_containers( T const & expected_container, T2 const & actual_container ) {
+			return ioHelper->compare_containers< T, T2 >( expected_container, actual_container );
```

Comment:
Why is there this level of indirection that seemingly does nothing but forward the call to IOHelper?  Is IOHelper inaccessible (ie private) or something?

Is this specifically for comparing an FCL type against a std type?  I assume if you just wanted to compare two std types you would use std::equal?  I bet you are using std::equal somewhere in the implementation. 

This seems like something universally useful beyond testing or io.  Maybe something for FCL directly?